substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/AlignedHeapChunk.java

```diff
@@ -263,6 +263,19 @@ public static void dirtyCardForObjectOfAlignedHeapChunk(Object object, boolean v
         }
     }
 
+    /**
+     * Dirty the card corresponding to the given Object without verify.
+     *
+     * This has to be fast, because it is used by the post-write barrier.
+     */
+    public static void dirtyCardForObjectOfAlignedHeapChunk(Object object) {
```

Comment:
Yes, it seems duplicate code here. We want to keep one method too.
But if we call this method at runtime, will get exception of the `AssertionNode` at the image build time.

What's your suggestion about this. : )

below is the exception:
`
Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing org.graalvm.compiler.replacements.nodes.AssertionNode.assertion(boolean, boolean, java.lang.String, java.lang.Object, java.lang.Object)
Parsing context:
	parsing org.graalvm.compiler.replacements.nodes.AssertionNode.assertion(AssertionNode.java:119)
	parsing com.oracle.svm.core.genscavenge.AlignedHeapChunk.dirtyCardForObjectOfAlignedHeapChunk(AlignedHeapChunk.java:261)
`

Reply:
You can avoid the error (and therefore the code duplication) if you change the call to `AssertionNode.assertion(...)` from
`AssertionNode.assertion(false, CardTable.isDirtyEntryAtIndexUnchecked(cardTableStart, index), "card must be dirty");`
to
`AssertionNode.assertion(false, CardTable.isDirtyEntryAtIndexUnchecked(cardTableStart, index), "card must be dirty", "", "");`

Roughly the same change can also be done for the unaligned chunks to avoid the code duplication there as well.